### PR TITLE
uint: make const matching work again

### LINKED ISSUE #420: [uint] const matching doesn't work anymore
Apparently, in order to match on a const value, `PartialEq` and `Eq` must be derived:
https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=c371a6f8796511d84a21186cbb4ea352

So #390 was a breaking change that broke some code (https://github.com/Conflux-Chain/conflux-rust/issues/1744, https://github.com/openethereum/openethereum/issues/11851).

As a quickfix we can revert this change and release a new version.

### PR DESCRIPTION
**Pull request by ordian**
Fixes #420.

Release scheduled for Wednesday, 12th.

### CODE REVIEW

File: uint/src/uint.rs [hunk 1]
```diff
@@ -445,7 +445,7 @@ macro_rules! construct_uint {
 		/// Little-endian large integer type
 		#[repr(C)]
 		$(#[$attr])*
-		#[derive(Copy, Clone)]
+		#[derive(Copy, Clone, Eq, PartialEq, Hash)]
```
Comment (niklasad1) [inline uint/src/uint.rs:448]: so, this won't work with length bigger than 32, correct? Is that a problem?

Comment (ordian) [reply uint/src/uint.rs:448]: correct, #388 is still unresolved, we'd need to figure out how to refactor our use of `unroll!` or the macro itself in order to fix that besides the 32-length limitation

But the 32-length limitation could be lifted soon with the stabilization of (subset of) const generics.

Comment (ordian) [reply uint/src/uint.rs:448]: https://github.com/rust-lang/rust/pull/74060

Comment (niklasad1) [reply uint/src/uint.rs:448]: ok, I see

**Comment by ordian** [comment]
Published from the branch.

**Review by dvdplm — APPROVED** [review]

**Pull request merged**